### PR TITLE
Add Search UI

### PR DESCRIPTION
**Pull request by LunarX**
Depends on #624 

Is missing:
* Wrong empty state illustration for search results
* Pagination
* Swipe refresh on search results to query again
* Swipe actions
* History is implemented in the next PR

Observation:
* There's no ui indication if the search is offline/online ?

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/main/menu/MoveViewModel.kt [hunk 1]
```diff
@@ -20,13 +20,19 @@ package com.infomaniak.mail.ui.main.menu
...
+import com.infomaniak.mail.data.cache.mailboxContent.MessageController
+import com.infomaniak.mail.data.cache.mailboxContent.ThreadController
 import kotlinx.coroutines.Dispatchers
 
 class MoveViewModel : ViewModel() {
 
     private val coroutineContext = viewModelScope.coroutineContext + Dispatchers.IO
 
-    val currentFolders = liveData(coroutineContext) { emit(FolderController.getRootsFolders().getMenuFolders()) }
+    fun message(messageUid: String) = liveData(coroutineContext) {
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/menu/MoveViewModel.kt:31]: rename to `getFolderIdByMessage`

File: app/src/main/java/com/infomaniak/mail/ui/main/menu/MoveViewModel.kt [hunk 2]
```diff
@@ -20,13 +20,19 @@ package com.infomaniak.mail.ui.main.menu
...
+        emit(MessageController.getMessage(messageUid)!!.folderId)
+    }
+
+    fun thread(threadUid: String) = liveData(coroutineContext) {
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/menu/MoveViewModel.kt:35]: rename to `getFolderIdByThread`

File: app/src/main/res/layout/item_search_folder.xml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        android:textColor="@color/primaryTextColor"
+        app:cornerRadius="0dp"
+        app:iconPadding="@dimen/alternativeMargin"
+        app:iconSize="18dp"
+        app:iconTint="?attr/colorPrimary"
+        tools:icon="@drawable/ic_archive_folder"
+        tools:text="Boîte de réception" />
+
+</LinearLayout>
+
```
Comment (KevinBoulongne) [inline app/src/main/res/layout/item_search_folder.xml:47]:
```suggestion
```

File: app/src/main/res/layout/fragment_search.xml [hunk 4]
```diff
@@ -15,20 +15,163 @@
...
+                android:layout_gravity="center_vertical"
+                android:visibility="gone"
+                app:description="@string/searchNoResultsDescription"
+                app:icon="@drawable/ic_search"
+                app:title="@string/searchNoResultsTitle"
+                tools:visibility="visible" />
+        </FrameLayout>
+    </androidx.swiperefreshlayout.widget.SwipeRefreshLayout>
+</androidx.coordinatorlayout.widget.CoordinatorLayout>
+
```
Comment (KevinBoulongne) [inline app/src/main/res/layout/fragment_search.xml:177]:
```suggestion
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

**Review by JorisBodin — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt [hunk 5]
```diff
@@ -334,6 +338,24 @@ class ThreadListAdapter(
...
+                        draftLocalUuid = it.draftLocalUuid,
+                        draftResource = it.draftResource,
+                        messageUid = it.messageUid,
+                    ).toBundle(),
+                )
+            }
+        } else {
+            safeNavigate(R.id.threadFragment, ThreadFragmentArgs(thread.uid).toBundle())
+        }
+    }
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt:357]: Put this function in Extension.kt

**Review by FabianDevel — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/search/SearchFragment.kt [hunk 6]
```diff
@@ -75,21 +125,115 @@ class SearchFragment : Fragment() {
...
-    private fun observeFolders() {
-        searchViewModel.folders.observe(viewLifecycleOwner) {
-            // TODO: handle folders ui
+    private fun setUi() = with(binding) {
+        toolbar.setNavigationOnClickListener { findNavController().popBackStack() }
+        swipeRefreshLayout.isEnabled = false
+
+        val popupMenu = createPopupMenu()
+
+        folderDropDown.setOnClickListener { popupMenu.show() }
```
[the remark this answers is not in the document]
Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/ui/main/search/SearchFragment.kt:134]: It creates it once when setting the ui and then shows it as many times as necessary when clicked

[the remark this answers is not in the document]
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/mail/ui/main/search/SearchFragment.kt:135]: true, I read too fast

**Review by FabianDevel — APPROVED** [review]

**Review by JorisBodin — APPROVED** [review]

**Pull request merged**